iox-#743 Validation of request-response example (#1235)
eclipse-iceoryx/iceoryx: iceoryx_examples/request_response/README.md

the sequence.

We provide three examples, the very basic typed and untyped examples
and the most natural setup combining a server with a Listener and a client using
a WaitSet. Since you can find the general setup and functionality of the client

> and the server also in the Listener/WaitSet example, we will focus now on this.

Author: What is meant, is that we focus on the listener/waitset example since it contains all the important things from the other examples as well. Maybe I should rephrase it...

    .or_else([](auto& error) { std::cout << "Could not allocate Request! Error: " << error << std::endl; });
```

Once the request has been sent, we block and wait for samples to arrive. Then we
iterate over the notification vector to check if we were triggered from our client:
<!-- [geoffrey] [iceoryx_examples/request_response/client_cxx_waitset.cpp] [[wait and check if the client triggered]] -->
```cpp
auto notificationVector = waitset.timedWait(iox::units::Duration::fromSeconds(5));

for (auto& notification : notificationVector)
{
    if (notification->doesOriginateFrom(&client))
    {
        // ...
    }
}
```

The client receives the responses from the server using `take()`
and extracts the sequence id with `response.getResponseHeader().getSequenceId()`.
When the server response comes in the correct order, the received sum is stored

> in the `addend` so it can be used for the next request.

Reviewer: ```suggestion
in the `ctx` struct for usage in the next request.
```
